pylint/checkers/base_checker.py

```diff
@@ -201,3 +202,32 @@ class BaseTokenChecker(BaseChecker):
     def process_tokens(self, tokens):
         """Should be overridden by subclasses."""
         raise NotImplementedError()
+
+
+class _BasicChecker(BaseChecker):
+    __implements__ = interfaces.IAstroidChecker
+    name = "basic"
+
+
+class _NameCheckerBase(_BasicChecker):
```

Comment:
I think this should be in a name_* checker file and not inherit from ``_BasicChecker``.

Reply:
I agreed with @DanielNoord to use linear inheritance for this file. Please do not ask for contradicting changes.